### PR TITLE
Fix for Perm Delete

### PR DESCRIPTION
**Pull request by siminsavani-msft**

### CODE REVIEW

File: cmd/zc_traverser_blob.go [hunk 1]
```diff
@@ -377,10 +377,13 @@ func (t *blobTraverser) createStoredObjectForBlob(preprocessor objectMorpher, bl
...
 	)
 
-	object.blobSnapshotID = blobInfo.Snapshot
 	object.blobDeleted = blobInfo.Deleted
-	if blobInfo.VersionID != nil {
-		object.blobVersionID = *blobInfo.VersionID
+	if t.includeDeleted && t.includeSnapshot {
+		object.blobSnapshotID = blobInfo.Snapshot
+	} else if t.includeDeleted && t.includeVersion {
+		if blobInfo.VersionID != nil {
```
Comment (mohsha-msft) [inline cmd/zc_traverser_blob.go:384]: Nit: Can be included in the above else condition. Is this a regression?

Comment (siminsavani-msft) [reply cmd/zc_traverser_blob.go:384]: Yeah, it wasn't picked up by the CI unfortunately.

**Review by adreed-msft — APPROVED** [review]

**Comment by adreed-msft** [comment]
Ditto on testing.

**Comment by zezha-msft** [comment]
We should expand E2E test to include versioning enabled accounts, which are missing in our CI right now.

**Pull request merged**